ca/ca.go

```diff
@@ -280,7 +280,7 @@ func (ca *CAImpl) newCertificate(domains []string, ips []net.IP, key crypto.Publ
 		}
 	}
 
-	certNotAfter := time.Now().AddDate(0, 0, ca.certLifetime)
+	certNotAfter := time.Now().Add(time.Duration(ca.certValidityPeriod-1) * time.Second)
```

Comment:
Oh my goodness I didn't even notice that it does two separate calls to `time.Now()`! Obviously you don't need to fix this now but this should definitely be `certNotBefore.Add(...)`

Reply:
Hm, that's a sharp eye, I haven't even looked at `certNotBefore` :stuck_out_tongue: